### ISSUE TITLE
How to train Chinese tokenizer using Syntaxnet?

### ISSUE THREAD

**Issue by i-xying** [dsindex/syntaxnet#15]
Hi , I have trained my Chinese model refering to how to train English model. But how can I make Chinese tokenizer? When I test my Chinese model with MODEL_DIRECTORY=/xy/models/syntaxnet/Chinesemodel
echo '這樣的處理也衍生了一些問題.' | ./parse.sh The result is just all the sentence as a noun.

**Comment by dsindex** [comment]
what about using parsey's cousins? 

https://github.com/tensorflow/models/blob/master/syntaxnet/universal.md

```
 MODEL_DIRECTORY=/where/you/unzipped/the/model/files
  cat sentences.txt | syntaxnet/models/parsey_universal/tokenize_zh.sh \
    $MODEL_DIRECTORY > output.conll
```

you can preprocess input sentences and feed to `./parse.sh`

beside using the pre-trained segmentation model for Chinese, 
there is no officially known method to train a segmentation model using the Syntaxnet. 

however, i guess some tricks.

- convert your training data into character-based.
(sorry for Korean example)
```
1	프랑스	프랑스	NNP	NNP	_	2	_	_	_
2	의	의	JKG	JKG	_	8	_	_	_
3	세계	세계	NNG	NNG	_	4	_	_	_
4	적	적	XSN	XSN	_	5	_	_	_
5	이	이	VCP	VCP	_	6	_	_	_
6	ᆫ	ᆫ	ETM	ETM	_	8	_	_	_
7	의상	의상	NNG	NNG	_	8	_	_	_
8	디자이너	디자이너	NNG	NNG	_	10	_	_	_
9	엠마누엘	엠마누엘	NNP	NNP	_	10	_	_	_
10	웅가로	웅가로	NNP	NNP	_	11	_	_	_
11	가	가	JKS	JKS	_	18	_	_	_
12	실내	실내	NNG	NNG	_	13	_	_	_
13	장식	장식	NNG	NNG	_	14	_	_	_
14	용	용	XSN	XSN	_	15	_	_	_
15	직물	직물	NNG	NNG	_	16	_	_	_
16	디자이너	디자이너	NNG	NNG	_	17	_	_	_
17	로	로	JKB	JKB	_	18	_	_	_
18	나서	나서	VV	VV	_	19	_	_	_
19	었	었	EP	EP	_	20	_	_	_
20	다	다	EF	EF	_	21	_	_	_
21	.	.	SF	SF	_	0	_	_	_

=> 

1	프	프	NNP	NNP	_	2	_	_	_
2	랑	랑	NNP	NNP	_	3	_	_	_
3	스	스	NNP	NNP	_	4	_	_	_
4	의	의	JKG	JKG	_	12	_	_	_
5	세	세	NNG	NNG	_	6	_	_	_
6	계	계	NNG	NNG	_	7	_	_	_
7	적	적	XSN	XSN	_	8	_	_	_
8	이	이	VCP	VCP	_	9	_	_	_
9	ᆫ	ᆫ	ETM	ETM	_	12	_	_	_
10	의	의	NNG	NNG	_	11	_	_	_
11	상	상	NNG	NNG	_	12	_	_	_
12	디	디	NNG	NNG	_	13	_	_	_
13	자	자	NNG	NNG	_	14	_	_	_
14	이	이	NNG	NNG	_	15	_	_	_
15	너	너	NNG	NNG	_	20	_	_	_
16	엠	엠	NNP	NNP	_	17	_	_	_
17	마	마	NNP	NNP	_	18	_	_	_
18	누	누	NNP	NNP	_	19	_	_	_
19	엘	엘	NNP	NNP	_	20	_	_	_
20	웅	웅	NNP	NNP	_	21	_	_	_
21	가	가	NNP	NNP	_	22	_	_	_
22	로	로	NNP	NNP	_	23	_	_	_
23	가	가	JKS	JKS	_	36	_	_	_
24	실	실	NNG	NNG	_	25	_	_	_
25	내	내	NNG	NNG	_	26	_	_	_
26	장	장	NNG	NNG	_	27	_	_	_
27	식	식	NNG	NNG	_	28	_	_	_
28	용	용	XSN	XSN	_	29	_	_	_
29	직	직	NNG	NNG	_	30	_	_	_
30	물	물	NNG	NNG	_	31	_	_	_
31	디	디	NNG	NNG	_	32	_	_	_
32	자	자	NNG	NNG	_	33	_	_	_
33	이	이	NNG	NNG	_	34	_	_	_
34	너	너	NNG	NNG	_	35	_	_	_
35	로	로	JKB	JKB	_	36	_	_	_
36	나	나	VV	VV	_	37	_	_	_
37	서	서	VV	VV	_	38	_	_	_
38	었	었	EP	EP	_	39	_	_	_
39	다	다	EF	EF	_	40	_	_	_
40	.	.	SF	SF	_	0	_	_	_
``` 

basically we want to segment input string to meaningful words. 
it seems that above data is not proper to segmentation. 

- set governor's index using word boundary itself.
```
1	프	프	NNP	NNP	_	2	_	_	_
2	랑	랑	NNP	NNP	_	3	_	_	_
3	스	스	NNP	NNP	_	4	_	_	_
4	의	의	JKG	JKG	_	40	_	_	_
5	세	세	NNG	NNG	_	6	_	_	_
6	계	계	NNG	NNG	_	7	_	_	_
7	적	적	XSN	XSN	_	8	_	_	_
8	이	이	VCP	VCP	_	9	_	_	_
9	ᆫ	ᆫ	ETM	ETM	_	40	_	_	_
10	의	의	NNG	NNG	_	11	_	_	_
11	상	상	NNG	NNG	_	40	_	_	_
12	디	디	NNG	NNG	_	13	_	_	_
13	자	자	NNG	NNG	_	14	_	_	_
14	이	이	NNG	NNG	_	15	_	_	_
15	너	너	NNG	NNG	_	40	_	_	_
16	엠	엠	NNP	NNP	_	17	_	_	_
17	마	마	NNP	NNP	_	18	_	_	_
18	누	누	NNP	NNP	_	19	_	_	_
19	엘	엘	NNP	NNP	_	40	_	_	_
20	웅	웅	NNP	NNP	_	21	_	_	_
21	가	가	NNP	NNP	_	22	_	_	_
22	로	로	NNP	NNP	_	23	_	_	_
23	가	가	JKS	JKS	_	40	_	_	_
24	실	실	NNG	NNG	_	25	_	_	_
25	내	내	NNG	NNG	_	40	_	_	_
26	장	장	NNG	NNG	_	27	_	_	_
27	식	식	NNG	NNG	_	28	_	_	_
28	용	용	XSN	XSN	_	40	_	_	_
29	직	직	NNG	NNG	_	30	_	_	_
30	물	물	NNG	NNG	_	40	_	_	_
31	디	디	NNG	NNG	_	32	_	_	_
32	자	자	NNG	NNG	_	33	_	_	_
33	이	이	NNG	NNG	_	34	_	_	_
34	너	너	NNG	NNG	_	35	_	_	_
35	로	로	JKB	JKB	_	40	_	_	_
36	나	나	VV	VV	_	37	_	_	_
37	서	서	VV	VV	_	38	_	_	_
38	었	었	EP	EP	_	39	_	_	_
39	다	다	EF	EF	_	40	_	_	_
40	.	.	SF	SF	_	0	_	_	_
```

here every boundary characters are governed by the last character.
and notice that we can't use the part-of-speech tag.(just ignore those fields)

if we get a model trained by this data, we can segment input sentence by identifying governor's index. 
```
insert space if governor's index is last one
do not insert space otherwise
```

does it seems to make sense?

**Comment by i-xying** [comment]
@dsindex Thank you very much ! I have used the parsey's cousin as you mentioned above . When I use the Chinese (download from the official website of TensorFlow) It can successfully segment sentences and give true POS and Parsey. But when I train my own Chinesemodel using the datasets UD_Chinese (Following your method in https://github.com/dsindex/syntaxnet),  It can not generate the tokenizer. The contrast of Chinese and Chinesemodel is as follows.
Chinese:
![image](https://cloud.githubusercontent.com/assets/24712951/21558667/9d368e6e-ce78-11e6-893f-03ee80139604.png)

Chinesemodel:
![image](https://cloud.githubusercontent.com/assets/24712951/21558671/b15bd7fa-ce78-11e6-97a1-0bf6f905d041.png)
When I train my own model , I used your giving bash--train.sh, and the context.pbtxt copying from the UD_English. I think I should transform the bash and the context.pbtxt , but I have no ideas about it. Is there any better suggestions? 
your giving suggestion I think it may very helpful when I create my own corpus, Thanks a lot.
